saw/Cryptol.sawcore

```diff
@@ -1250,6 +1250,16 @@ ecSle : (n : Num) -> seq n Bool -> seq n Bool -> Bool;
 ecSle =
   finNumRec (\ (n : Num) -> seq n Bool -> seq n Bool -> Bool) bvsle;
 
+-- Array operations
+ecArrayConstant : (a b : sort 0) -> b -> Array a b;
+ecArrayConstant = arrayConstant;
+
+ecArrayLookup : (a b : sort 0) -> (Array a b) -> a -> b;
+ecArrayLookup = arrayLookup;
+
+ecArrayUpdate : (a b : sort 0) -> (Array a b) -> a -> b -> (Array a b);
+ecArrayUpdate = arrayUpdate;
+
```

Comment:
Are these definitions necessary? Could we just refer to the Prelude functions from the translation?

Reply:
Since the types are exactly the same, I guess we could, but the extra indirection doesn't really cost much. For consistency it is kind of nice to have all the cryptol primitives defined explicitly in `Cryptol.sawcore`. And also it makes it easier if we ever change the type of the cryptol primitives even slightly (for example, if we wanted to add a class constraint) because in that case the indirection would become necessary.

Reply:
Yeah, fair point. I have no real objection to leaving them in.

Reply:
I agree with both of you guys. Initially, I wanted to take the approach that @atomb suggested, and then I changed my mind for the reasons that @brianhuffman mentioned. Are you guys my subconscious? :-)